src/common/gui/CLFOGui.cpp

```diff
@@ -974,6 +974,10 @@ CMouseEventResult CLFOGui::onMouseDown(CPoint& where, const CButtonState& button
    {
       if (ss && lfodata->shape.val.i == ls_stepseq)
       {
+         // find out if a microtuning is loaded and store the scale length for Alt-drag quantization to scale degrees
+         if (!storage->isStandardTuning && storage->currentScale.count > 1)
+            altQuantStep = storage->currentScale.count;
```

Comment:
Oh another bug. You made altquantstep a member so it will stick around, therefore if you make a 15 note tuning it gets set to 15, then if you untune it doesn't get set back to 12.

the trick is to *not* put it in the h but make it a local here and initialize it to 12. So remove the altQuantStep in the header file below and add the line `int altQuantStep = 12` right above this if.

Reply:
OK that's cool but then altQuantStep is not visible in ::onMouseDrag...

Reply:
It’s not a query it’s just a memory access. Really about 3-10 instructions. But you are correct you could do an if. 

It’s sort of a trade off (and I would merge either) 

But: More bugs come from state needing to be coherent than come from doing an extra dereference so if I were coding it I would just look up the value in each onMouseDrag. They happen outside the UI thread and already access the storage object so it’s not a performance problem. But again it would be correct in either case this way and I’m happy for either change, just sharing a purely philosophical outlet.

(You do need to do one or the other though otherwise the state hangs around; if it was a local that wouldn’t be a bug available to us)

Reply:
Alrighty! All done now :)